tests/unit/test_cmr_search.py

```diff
@@ -301,21 +340,52 @@ def test_download_granule(self, mock_requests_get):
         # Set the return_value of `mock_requests_get` to mock_response
         mock_requests_get.return_value = mock_response
         file_path = download_granule(link,
-                                     token="foo",
+                                     auth_header=self.bearer_token_header,
+                                     out_directory=self.output_dir)
+        # Check if `download_granule` was called once with expected parameters
+        mock_requests_get.assert_called_once_with(
+            link,
+            headers={'Authorization': self.bearer_token_header},
+            timeout=10
+        )
+        # Check if download file contains expected content from `requests.get`
+        with self.subTest('Test if downloaded file contains expected content'):
+            with open(file_path, 'r', encoding='utf-8') as file:
+                data = file.read()
+
+            self.assertEqual(data, link)
```

Comment:
Is the content of these downloads usually a link? if not then this is a little confusing.

Reply:
- [ ] Improve download tests.